framework/src/kernels/ADDiffusion.C

```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADDiffusion.h"
+
+registerADMooseObject("MooseApp", ADDiffusion);
+
+template <>
+InputParameters
+validParams<ADDiffusion<RESIDUAL>>()
+{
+  InputParameters p = validParams<ADKernel<RESIDUAL>>();
+  p.addClassDescription("Same as `Diffusion` in terms of physics/residual, but the Jacobian "
+                        "is computed using forward automatic differentiation");
+  return p;
+}
+template <>
+InputParameters
+validParams<ADDiffusion<JACOBIAN>>()
```

Comment:
I think we need to wrap this up in a macro

Reply:
To be clear - the macro should just declar both

Reply:
Looks like I missed `ADDiffusion`. I got to every other one I think